webapp/lib/controller.dart

```diff
@@ -90,6 +96,47 @@ void initUI() {
   );
 }
 
+void setupProjectTimer(String project) {
+  var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+  projectTimers[project] = timer;
+}
+
+model.NeedsReplyData getLatestDataForSelectedProject(List<model.NeedsReplyData> updatedData, String selectedProjectName) {
+  var latestProjectData = [];
+  latestProjectData.addAll(updatedData.where((data) => data.project == selectedProjectName));
+  if (latestProjectData.isEmpty) {
+    return null;
+  } else {
+    return latestProjectData.last;
+  }
+}
+
+void checkNeedsReplyMetricsStale(List<model.NeedsReplyData> updatedData) {
+  updatedData.sort((d1, d2) => d1.datetime.compareTo(d2.datetime));
+  
+  String selectedProjectName = view.contentView.projectSelectorView.selectedProject;
+  var latestSelectedProjectData = getLatestDataForSelectedProject(updatedData, selectedProjectName);
+
+  if (latestSelectedProjectData != null) {
+    var now = new DateTime.now();
+    int lastUpdateTimeDiff =  now.difference(latestSelectedProjectData.datetime).inHours;
+    
+    if (lastUpdateTimeDiff >= 1) {
+        view.contentView.stale = true;
+      } else {
+        view.contentView.stale = false;
+      }
+  } else {
+    view.contentView.stale = true;
+  }
```

Comment:
I think this remained from the previous algorithm, and I think it actually breaks up the logic now. If I understand this code correctly it would mark the current view as stale if a project other than the selected on has been updated, which isn't what we want. Can you double check what happens here, and remove the else if it's not needed anymore?

Reply:
This is supposed to mark the charts as stale if we haven't received any new data for the selected project